src/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -45,27 +45,37 @@ public PingTest(ITestOutputHelper output)
         }
 
         private void PingResultValidator(PingReply pingReply, IPAddress localIpAddress)
+        {
+            PingResultValidator(pingReply, new IPAddress[] { localIpAddress });
+        }
+
+        private void PingResultValidator(PingReply pingReply, IPAddress[] localIpAddresses)
         {
             if (pingReply.Status == IPStatus.TimedOut)
             {
                 // Workaround OSX ping6 bug, refer issue #15018
-                Assert.Equal(AddressFamily.InterNetworkV6, localIpAddress.AddressFamily);
+                Assert.Equal(AddressFamily.InterNetworkV6, pingReply.Address.AddressFamily);
                 Assert.True(RuntimeInformation.IsOSPlatform(OSPlatform.OSX));
                 return;
             }
 
             Assert.Equal(IPStatus.Success, pingReply.Status);
-            if (!pingReply.Address.Equals(localIpAddress))
+            foreach (IPAddress localIpAddress in localIpAddresses)
             {
-                // Test is going to fail. Collect some more info.
-                _output.WriteLine($"Reply address {pingReply.Address} is not expected {localIpAddress}");
-                IPHostEntry hostEntry = Dns.GetHostEntry(TestSettings.LocalHost);
-                foreach (IPAddress address in hostEntry.AddressList)
+                if (pingReply.Address.Equals(localIpAddress))
                 {
-                    _output.WriteLine($"Local address {address}");
+                    return;
                 }
```

Comment:
Nit: you could use LINQ to simplify cases like this / avoid the manual looping, e.g.
```C#
if (localIpAddresses.Any(addr => pingReply.Address.Equals(addr))
    return;
```

Reply:
We still should keep the "_output.WriteLine" for diagnostic purposes.